### PR TITLE
Optional Keypad Implemented Closes #87

### LINKED ISSUE #87: Optional KeyPad

### PR DESCRIPTION
**Pull request by gacosta1**
The optional keypad has been implemented.

### CODE REVIEW

**Review by r3cebarnett — APPROVED** [review]

**Review by gmross — APPROVED** [review]

File: Software/scan_key/AuthenticationSQL.py [hunk 1]
```diff
@@ -62,8 +61,34 @@ def twoFactorAuth(id, pin, cursor):
         return(False) #if pin is invalid, then not allowed
 
 def getID(idString, cursor):
-    cursor.execute("SELECT CUID, pin FROM USER WHERE t1String = " + idString) #getting user w/ the id
+    cursor.execute("SELECT CUID FROM USER WHERE t1String = " + idString) #getting user w/ the id
     data = cursor.fetchone() #fetching data into array
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py:65]: On line 10 you did some shenanigans to get currMachineID into the SQL query - why not here?

File: Software/scan_key/AuthenticationSQL.py [hunk 2]
```diff
@@ -62,8 +61,34 @@ def twoFactorAuth(id, pin, cursor):
...
-
     cuid = data[0] #and this is the cuid of the person with the id string
     return cuid
+
+def getMachineID():
+    currMachineID = subprocess.check_output(['hostname'])
+    currMachineID = currMachineID.strip()
+    currMachineID = (currMachineID).decode('UTF-8')
+    return(currMachineID)
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py:73]: Do our style conventions call for return() or just return? All the functions below this use return.

File: Software/scan_key/AuthenticationSQL.py [hunk 3]
```diff
@@ -62,8 +61,34 @@ def twoFactorAuth(id, pin, cursor):
...
+
+def getOPT(idString, cursor):
+    cursor.execute("SELECT user2fa FROM USER WHERE t1String = " + idString) #getting user w/ the opt$
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py:76]: See line 64

File: Software/scan_key/main.py [hunk 4]
```diff
@@ -19,14 +19,16 @@
 if __name__ == '__main__':
 
     # connecting to the SQL Server and Database
-    cnx = pymysql.connect(user='CATS', password='CATS', host='192.168.1.2', database='CATS', autocommit=True)
+    config = configparser.RawConfigParser() #instantiate config reader
+    config.read(os.path.join(os.path.abspath(os.path.dirname(__file__)), 'config.cfg')) #actually read the config file
```
Comment (ophilli) [inline Software/scan_key/main.py:23]: Lol that's some magic

**Review by ophilli — CHANGES_REQUESTED** [review]
Just a couple style comments really. I assume you tested this code - we really need unit testing . . .

File: Software/scan_key/AuthenticationSQL.py (hunk 1 above)
Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py:65]: There's something with the encoding of the string that won't allow it to be used in the SQL instruction so I had to use '%s'. Or I could do
" ' " + currMachineID + " ' "

File: Software/scan_key/AuthenticationSQL.py (hunk 2 above)
Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py:73]: I'll make sure to change the return values to have parentheses around them

**Review by ophilli — APPROVED** [review]

**Comment by ophilli** [comment]
Approved - do you still want @thorbrockman to review or should I merge?

**Comment by gacosta1** [comment]
@ophilli Merge please!

**Pull request merged**